testing/integration/logs_ingestion_test.go

```diff
@@ -127,6 +131,37 @@ func startMockES(t *testing.T) string {
 	return s.URL
 }
 
+// checkHealthAtStartup ensures all the beats and agent are healthy and working before we continue
+func checkHealthAtStartup(t *testing.T, ctx context.Context, agentFixture *atesting.Fixture) {
+	// because we need to separately fetch the PIDs, wait until everything is healthy before we look for running beats
+	compDebugName := ""
+	require.Eventually(t, func() bool {
+		allHealthy := true
+		status, err := agentFixture.ExecStatus(ctx)
+		if err != nil {
+			t.Logf("agent status returned an error: %v", err)
+			return false
+		}
+
+		for _, comp := range status.Components {
+			// make sure the components include the expected integrations
+			for _, v := range comp.Units {
+				t.Logf("unit ID: %s", v.UnitID)
+				t.Logf("unit state: %s", v.Message)
+				if v.State != int(cproto.State_HEALTHY) {
+					allHealthy = false
+				}
+			}
+			t.Logf("component state: %s", comp.Message)
+			if comp.State != int(cproto.State_HEALTHY) {
+				compDebugName = comp.Name
+				allHealthy = false
+			}
+		}
+		return allHealthy
```

Comment:
Nitpicking here: can we log the full status of the agent instead of having multiple separate logs per invocation? Something like:

```suggestion
		t.Logf("Received agent status:\n%+v\n", status) // this can be re-marshaled to JSON if we prefer that notation
		for _, comp := range status.Components {
			// make sure the components include the expected integrations
			for _, v := range comp.Units {
				if v.State != int(cproto.State_HEALTHY) {
					allHealthy = false
				}
			}
			if comp.State != int(cproto.State_HEALTHY) {
				compDebugName = comp.Name
				allHealthy = false
			}
		}
		return allHealthy
```